code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/MultiIconNotificationBuilder.kt

```diff
@@ -0,0 +1,131 @@
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.app.NotificationManager
+import android.content.Context
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.R
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewImage
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MultiIconPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+internal object MultiIconNotificationBuilder {
+    const val SELF_TAG = "MultiIconNotificationBuilder"
+
+    fun construct(
+        context: Context,
+        pushTemplate: MultiIconPushTemplate,
+        trackerActivityClass: Class<out Activity>?
+    ): NotificationCompat.Builder {
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an icon template push notification."
+        )
+
+        val packageName = context.packageName
+        val notificationLayout = RemoteViews(packageName, R.layout.push_template_multi_icon)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse: String = notificationManager.createNotificationChannelIfRequired(
+            context,
+            pushTemplate
+        )
+
+        populateIconsForMultiIconTemplate(
+            context,
+            trackerActivityClass,
+            notificationLayout,
+            pushTemplate,
+            pushTemplate.templateItemList,
+            packageName
+        )
+
+        setCancelIcon(
+            notificationLayout,
+            pushTemplate,
+            packageName)
+
+        notificationLayout.setRemoteViewClickAction(
+            context,
+            trackerActivityClass,
+            R.id.five_icon_close_button,
+            null,
+            null,
+            pushTemplate.tag,
+            false
+        )
+
+        return AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            notificationLayout,
+            notificationLayout,
+            R.id.carousel_container_layout
+        )
+    }
+
+    private fun setCancelIcon(
+        notificationLayout: RemoteViews,
+        pushTemplate: MultiIconPushTemplate,
+        packageName: String?
```

Comment:
package name is unused in this function.

Reply:
Done, removed.